src/components/film-info/film-info.tsx

```diff
@@ -1,28 +1,54 @@
-import { FilmInfo as FilmInfoType } from '@/types/film-info';
-import FilmNavigation from '../film-navigation/film-navigation';
+import { Film } from '@/types/film';
+import { FilmDetails as FilmDetailsType } from '@/types/film-details';
+import { FilmOverview as FilmOverviewType } from '@/types/film-overview';
 import FilmOverview from '../film-overview/film-overview';
+import FilmDetails from '../film-details/film-details';
+import UserReviews from '../user-reviews/user-reviews';
 
 type FilmInfoProps = {
-  filmInfo: FilmInfoType;
+  film: Film;
+  activeTab: string;
 };
 
-export default function FilmInfo({ filmInfo }: FilmInfoProps): JSX.Element {
+export default function FilmInfo({
+  film,
+  activeTab,
+}: FilmInfoProps): JSX.Element {
+  const {
+    director,
+    genre,
+    year,
+    reviews,
+    actors,
+    duration,
+    description,
+    ratingLevel,
+    rating,
+    ratingCount,
+  } = film;
+
+  const filmDetails: FilmDetailsType = {
+    director,
+    genre,
+    year,
+    actors,
+    duration,
+  };
+
+  const filmOverview: FilmOverviewType = {
+    rating,
+    ratingLevel,
+    ratingCount,
+    description,
+    actors,
+    director,
+  };
+
   return (
-    <div className="film-card__info">
-      <div className="film-card__poster film-card__poster--big">
-        <img src={filmInfo.previewImage} alt={filmInfo.name} width={218} height={327} />
-      </div>
-      <div className="film-card__desc">
-        <FilmNavigation />
-        <FilmOverview
-          filmRating={filmInfo.rating}
-          filmActors={filmInfo.actors}
-          filmDescription={filmInfo.description}
-          filmDirector={filmInfo.director}
-          filmRatingCount={filmInfo.ratingCount}
-          filmRatingLevel={filmInfo.ratingLevel}
-        />
-      </div>
-    </div>
+    <>
+      {activeTab === 'Overview' && <FilmOverview filmOverview={filmOverview} />}
+      {activeTab === 'Details' && <FilmDetails filmDetails={filmDetails} />}
+      {activeTab === 'Reviews' && <UserReviews reviews={reviews} />}
+    </>
```

Comment:
Я бы заменил на вызов отдельной функции и в ней switch
```suggestion
   const renderTabContent = () => {
     switch (activeTab: ActiveTabs): {
       case ActiveTabs.Details:
          ...................
         
     }
   }

    <>
      {renderTabContent()}
    </>
```